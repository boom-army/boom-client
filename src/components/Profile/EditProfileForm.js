import { useState } from 'react';
import Avatar from '@mui/material/Avatar';
import Button from '../../styles/Button';
import CoverPhoto from '../../styles/CoverPhoto';
import Form from '../../styles/Form';
import Input from '../Input';
import { TextareaAutosize } from "@mui/material";
import useInput from '../../hooks/useInput';
import { PROFILE, EDIT_PROFILE } from '../../queries/profile';
import { SIGN_FILE } from '../../queries/files';
import { displayError } from '../../utils';
import { uploadImage } from '../../utils';
import { useMutation } from '@apollo/client';
import { useSnackbar } from 'notistack';
import { useNavigate } from 'react-router-dom';

export const EditProfileForm = ({ profile, setUser }) => {
  const navigate = useNavigate();
  const [avatarState, setAvatar] = useState('');
  const [coverPhotoState, setCoverPhoto] = useState('');

  const handle = useInput(profile && profile.handle);
  const consumerName = useInput(profile && profile.consumerName);
  const location = useInput(profile && profile.location);
  const website = useInput(profile && profile.website);
  const dob = useInput(profile && profile.dob);
  const avatar = useInput(profile && profile.avatar);
  const bio = useInput(profile && profile.bio);
  const coverPhoto = useInput(profile && profile.coverPhoto);

  const [editProfileMutation, { loading }] = useMutation(EDIT_PROFILE, {
    refetchQueries: [{ query: PROFILE, variables: { handle: handle.value } }],
  });
  const [signFileMutation] = useMutation(SIGN_FILE);

  const { enqueueSnackbar } = useSnackbar();

  const handleEditProfile = async (e) => {
    e.preventDefault();

    if (!consumerName.value) {
      return enqueueSnackbar('You cannot leave name empty', {
        variant: 'error',
      });
    }

    try {
      const { data } = await editProfileMutation({
        variables: {
          handle: handle.value,
          consumerName: consumerName.value,
          dob: dob.value,
          bio: bio.value,
          location: location.value,
          website: website.value,
          avatar: avatarState ? avatarState : avatar.value,
          coverPhoto: coverPhotoState ? coverPhotoState : coverPhoto.value,
        },
      });

      setUser({
        handle: data.editProfile.handle,
        publicAddress: data.editProfile.publicAddress,
        id: data.editProfile.id,
      });

      navigate(`/${data.editProfile.handle}`);

      enqueueSnackbar('Your profile has been updated ðŸ¥³.', {
        variant: 'success',
      });
    } catch (err) {
      return displayError(err, enqueueSnackbar);
    }
  };

  const handleCoverPhoto = async (e) => {
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadImage(file, signedUrl, enqueueSnackbar);
      const imageUrl = imageData.config.url.split('?')[0];
      setCoverPhoto(imageUrl);
    } catch (error) {
      console.log(error);
    }
  };

  const handleAvatar = async (e) => {
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadImage(file, signedUrl, enqueueSnackbar);
      const imageUrl = imageData.config.url.split('?')[0];
      setAvatar(imageUrl);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <Form lg onSubmit={handleEditProfile}>
      <div className="cover-photo">
        <label htmlFor="cover-photo-input">
          <CoverPhoto
            src={coverPhotoState ? coverPhotoState : coverPhoto.value}
            alt="cover"
          />
        </label>
        <input
          type="file"
          id="cover-photo-input"
          accept="image/*"
          onChange={handleCoverPhoto}
        />
      </div>

      <div className="avatar-input">
        <label htmlFor="avatar-input-file">
          <Avatar src={avatarState ? avatarState : avatar.value} />
        </label>
        <input
          type="file"
          accept="image/*"
          id="avatar-input-file"
          onChange={handleAvatar}
        />
      </div>

      <Input
        lg={true}
        text="Handle"
        value={handle.value}
        onChange={handle.onChange}
      />

      <Input
        lg={true}
        text="Name"
        value={consumerName.value}
        onChange={consumerName.onChange}
      />
      <div className="bio-wrapper">
        <label className="bio" htmlFor="bio">
          Bio
        </label>
        <TextareaAutosize
          id="bio"
          placeholder="Bio"
          value={bio.value}
          onChange={bio.onChange}
        />
      </div>
      <Input
        lg={true}
        text="Website"
        value={website.value}
        onChange={website.onChange}
      />
      <Input
        lg={true}
        text="Date of Birth"
        value={dob.value}
        onChange={dob.onChange}
      />
      <Input
        lg={true}
        text="Location"
        value={location.value}
        onChange={location.onChange}
      />
      <Button outline disabled={loading} type="submit">
        {loading ? 'Saving' : 'Save'}
      </Button>
    </Form>
  );
};
