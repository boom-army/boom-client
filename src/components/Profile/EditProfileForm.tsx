import AddPhotoAlternateIcon from "@mui/icons-material/AddPhotoAlternate";
import {
  Avatar,
  Button,
  Box,
  Container,
  TextField,
  Grid,
  Badge,
  Input,
} from "@mui/material";
import { SIGN_FILE } from "../../queries/files";
import { useTheme } from '@mui/material/styles';
import { cleanTypeName, displayError, uploadFile } from "../../utils";
import {
  ProfileDocument,
  ProfileQuery,
  useEditProfileMutation,
} from "../../generated/graphql";
import { User } from "../../contexts/user";
import { useInput } from "../../hooks/useInput";
import { useMutation } from "@apollo/client";
import { useNavigate } from "react-router-dom";
import { useSnackbar } from "../../contexts/snackbar";
import { useState, useContext } from "react";
import { UserAvatar } from "../UserAvatar";

interface Profile {
  profile: ProfileQuery["profile"]
  setUser: (user: User | null) => void | User 
}

export const EditProfileForm = ({ profile, setUser }: Profile) => {
  const theme = useTheme();
  const navigate = useNavigate();

  const handle = useInput(profile.handle);
  const consumerName = useInput(profile.consumerName as string);
  const location = useInput(profile.location as string);
  const website = useInput(profile.website as string);
  const dob = useInput(profile.dob as string);
  const avatar = useInput(profile.avatar);
  const bio = useInput(profile.bio as string);
  const coverPhoto = useInput(profile.coverPhoto as string);

  const [avatarState, setAvatar] = useState(avatar?.value ?? "");
  const [dataState, setData] = useState(cleanTypeName(profile.data) ?? {});
  const [coverPhotoState, setCoverPhoto] = useState(
    coverPhoto?.value ?? "/assets/default-cover.png"
  );

  const [editProfileMutation, { loading }] = useEditProfileMutation({
    refetchQueries: [
      { query: ProfileDocument, variables: { handle: handle.value } },
    ],
  });
  const [signFileMutation] = useMutation(SIGN_FILE);

  const { enqueueSnackbar } = useSnackbar();

  const handleEditProfile = async (e: any) => {
    e.preventDefault();

    if (!consumerName.value) {
      return enqueueSnackbar("You cannot leave name empty", {
        variant: "error",
      });
    }

    try {
      const { data }: any = await editProfileMutation({
        variables: {
          handle: handle.value,
          consumerName: consumerName.value,
          dob: dob.value,
          bio: bio.value,
          location: location.value,
          website: website.value,
          avatar: avatarState,
          data: dataState,
          coverPhoto: coverPhotoState,
        },
      });

      setUser({
        handle: data.editProfile.handle,
        publicAddress: data.editProfile.publicAddress,
        id: data.editProfile.id,
      });

      navigate(`/${data.editProfile.handle}`);

      enqueueSnackbar("Your profile has been updated ðŸ¥³.", {
        variant: "success",
      });
    } catch (err) {
      return displayError(err, enqueueSnackbar);
    }
  };

  const handleCoverPhoto = async (e: any) => {
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadFile(file, signedUrl, enqueueSnackbar);
      const imageUrl: string | undefined | any =
        imageData?.config?.url?.split("?")[0];
      setCoverPhoto(imageUrl);
    } catch (error) {
      console.log(error);
      displayError(error, enqueueSnackbar);
    }
  };

  const handleAvatar = async (e: any) => {
    console.log('boom');
    
    try {
      const file = e.target.files[0];
      const { data } = await signFileMutation({
        variables: {
          file: file.name,
          type: file.type,
        },
      });
      const signedUrl = data.signFileUrl;
      const imageData = await uploadFile(file, signedUrl, enqueueSnackbar);
      const imageUrl: string | undefined | any =
        imageData?.config?.url?.split("?")[0];
      // set avatar state to NOT NFT
      const userData = cleanTypeName(profile.data);
      delete userData.avatarMint;
      delete userData.avatarUpdateAuthority;
      setData(userData);
      // set avatar
      setAvatar(imageUrl);
    } catch (error) {
      console.log(error);
      displayError(error, enqueueSnackbar);
    }
  };

  return (
    <>
      <Container>
        <Box
          component="form"
          noValidate
          sx={{ mt: 3 }}
          onSubmit={handleEditProfile}
        >
          <Grid container spacing={2}>
            <Box
              position="relative"
              mb={2}
              sx={{
                width: "100%",
                height: "200px",
                backgroundImage: `url("${coverPhotoState}")`,
                backgroundPosition: "center center",
                backgroundSize: "cover",
                "&:hover": {
                  opacity: "80%",
                },
              }}
            >
              <Box
                sx={{
                  position: "absolute",
                  top: "40%",
                  left: "45%",
                }}
              >
                <label htmlFor="cover-photo-input">
                  <Avatar
                    sx={{
                      width: 30,
                      height: 30,
                      backgroundColor: theme.palette.primary.main,
                      cursor: "pointer",
                    }}
                  >
                    <AddPhotoAlternateIcon sx={{ width: 22, height: 22 }} />
                  </Avatar>
                  <Input
                    id="cover-photo-input"
                    type="file"
                    inputProps={{ accept: "image/*" }}
                    sx={{ display: "none" }}
                    onChange={handleCoverPhoto}
                  />
                </label>
              </Box>
              <Box
                sx={{
                  position: "absolute",
                  bottom: "1em",
                  left: "1em",
                }}
              >
                <label htmlFor="avatar-input-file">
                  <Input
                    id="avatar-input-file"
                    type="file"
                    inputProps={{ accept: "image/*" }}
                    sx={{ display: "none" }}
                    onChange={handleAvatar}
                  />
                  <Badge
                    overlap="circular"
                    anchorOrigin={{ vertical: "bottom", horizontal: "right" }}
                    badgeContent={
                      <Avatar
                        sx={{
                          width: 22,
                          height: 22,
                          border: `2px solid ${theme.palette.background.default}`,
                          backgroundColor: theme.palette.primary.main,
                        }}
                      >
                        <AddPhotoAlternateIcon sx={{ width: 14, height: 14 }} />
                      </Avatar>
                    }
                  >
                    <UserAvatar
                      avatar={avatarState}
                      isNFT={dataState.avatarMint}
                      sx={{
                        cursor: "pointer",
                      }}
                    />
                  </Badge>
                </label>
              </Box>
            </Box>
            <Grid item xs={12}>
              <TextField
                name="handle"
                required
                fullWidth
                autoFocus
                id="handle"
                label="Handle"
                InputLabelProps={{
                  shrink: true,
                  style: { color: theme.palette.secondary.main },
                }}
                InputProps={{
                  style: { color: theme.palette.secondary.main },
                }}
                value={handle.value}
                onChange={handle.onChange}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                name="name"
                required
                fullWidth
                autoFocus
                id="name"
                label="Name"
                InputLabelProps={{
                  shrink: true,
                  style: { color: theme.palette.secondary.main },
                }}
                InputProps={{
                  style: { color: theme.palette.secondary.main },
                }}
                value={consumerName.value}
                onChange={consumerName.onChange}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                name="bio"
                required
                fullWidth
                autoFocus
                rows={4}
                id="bio"
                label="Bio"
                InputLabelProps={{
                  shrink: true,
                  style: { color: theme.palette.secondary.main },
                }}
                InputProps={{
                  style: { color: theme.palette.secondary.main },
                }}
                value={bio.value}
                onChange={bio.onChange}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                name="website"
                required
                fullWidth
                autoFocus
                id="website"
                label="Website"
                InputLabelProps={{
                  shrink: true,
                  style: { color: theme.palette.secondary.main },
                }}
                InputProps={{
                  style: { color: theme.palette.secondary.main },
                }}
                value={website.value}
                onChange={website.onChange}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                name="dob"
                required
                fullWidth
                autoFocus
                id="dob"
                label="Date of Birth"
                InputLabelProps={{
                  shrink: true,
                  style: { color: theme.palette.secondary.main },
                }}
                InputProps={{
                  style: { color: theme.palette.secondary.main },
                }}
                value={dob.value}
                onChange={dob.onChange}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                name="location"
                required
                fullWidth
                autoFocus
                id="location"
                label="Location"
                InputLabelProps={{
                  shrink: true,
                  style: { color: theme.palette.secondary.main },
                }}
                InputProps={{
                  style: { color: theme.palette.secondary.main },
                }}
                value={location.value}
                onChange={location.onChange}
              />
            </Grid>
            <Grid item xs={12}>
              <Button
                sx={{ mt: 3, mb: 2 }}
                fullWidth
                variant="contained"
                type="submit"
                size="large"
                disabled={loading}
              >
                {loading ? "Saving" : "Save profile"}
              </Button>
            </Grid>
          </Grid>
        </Box>
      </Container>
    </>
  );
};
