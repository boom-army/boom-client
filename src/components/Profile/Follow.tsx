import React, { useState } from "react";
import PersonAddIcon from "@mui/icons-material/PersonAdd";
import PersonRemoveIcon from "@mui/icons-material/PersonRemove";
import { Box, Link } from "@mui/material";
import { USER_FOLLOW } from "../../queries/follow";
import { FOLLOW, UNFOLLOW } from "../../queries/follow";
import { displayError } from "../../utils";
import { useMutation } from "@apollo/client";
import { useSnackbar } from "../../contexts/snackbar";
import { FeedDocument } from "../../generated/graphql";

interface FollowProps {
  isFollowing: boolean;
  id: string;
  sm?: boolean;
  relative?: boolean;
}

export const Follow = ({
  isFollowing,
  id,
  sm = false,
  relative = false,
}: FollowProps) => {
  const [followState, setFollowState] = useState(isFollowing);

  const [followMutation] = useMutation(FOLLOW, {
    variables: { id },
    refetchQueries: [{ query: FeedDocument }, { query: USER_FOLLOW }],
  });

  const [unfollowMutation] = useMutation(UNFOLLOW, {
    variables: { id },
    refetchQueries: [{ query: FeedDocument }, { query: USER_FOLLOW }],
  });

  const { enqueueSnackbar } = useSnackbar();

  const handleFollow = async () => {
    if (followState) {
      setFollowState(false);
      try {
        await unfollowMutation();
      } catch (err) {
        displayError(err, enqueueSnackbar);
      }
    } else {
      setFollowState(true);
      try {
        await followMutation();
      } catch (err) {
        displayError(err, enqueueSnackbar);
      }
    }
  };

  return (
    <Box>
      <Link onClick={handleFollow} sx={{ cursor: "pointer" }}>
        {followState ? <PersonRemoveIcon /> : <PersonAddIcon />}
      </Link>
    </Box>
  );
};
