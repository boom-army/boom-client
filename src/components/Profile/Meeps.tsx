import React from "react";
import { Box, Typography } from "@mui/material";
import { ShowTweet } from "../Tweet";
import { Tweet, ProfileQuery } from "../../generated/graphql";

export const Meeps: React.FC<{
  data: ProfileQuery | undefined;
}> = ({ data }) => {
  return (
    <>
      {data && data.profile && data.profile.tweets && data.profile.tweets.length
        ? data.profile.tweets.map((tweet) => (
            <ShowTweet
              key={tweet.id}
              tweet={tweet as Tweet}
              offset={10}
              parentTweetId=""
            />
          ))
        : (
          <Box pt={10} sx={{ display: "flex", justifyContent: "center" }}>
            <Typography>You haven't Meeped yet</Typography>
          </Box>
        )}
    </>
  );
};
