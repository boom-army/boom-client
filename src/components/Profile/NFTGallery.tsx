import Clear from "@mui/icons-material/Clear";
import DoNotDisturbOnIcon from "@mui/icons-material/DoNotDisturbOn";
import React, { useContext, useState, useMemo, useEffect } from "react";
import {
  Box,
  Button,
  Chip,
  IconButton,
  Link,
  Modal,
  Stack,
  Typography,
} from "@mui/material";
import { Loader } from "../Loader";
import {
  Metadata,
  MetadataData,
} from "@metaplex-foundation/mpl-token-metadata";
import { ThemeContext } from "../../contexts/theme";
import { currentCluster } from "../../utils/utils";
import { displayError } from "../../utils";
import { useConnection } from "@solana/wallet-adapter-react";
import {
  useEditProfileMutation,
  ProfileDocument,
  ProfileQuery,
} from "../../generated/graphql";
import { useSnackbar } from "../../contexts/snackbar";
import { useParams } from "react-router-dom";

interface NFTTileProps {
  data: MetadataData;
  cluster: string;
}

interface NFTTraits {
  trait_type: string;
  value: string;
}

interface URIData {
  name: string;
  description: string;
  image: string;
  attributes: NFTTraits[];
}

const NFTTile: React.FC<NFTTileProps> = ({ data, cluster }) => {
  const { theme } = useContext(ThemeContext);
  const { enqueueSnackbar } = useSnackbar();
  let { handle } = useParams<string>();

  const [nftSelect, toggleNftSelect] = useState(false);
  const [uRIData, setURIData] = useState<URIData>();

  const [editProfileMutation] = useEditProfileMutation({
    refetchQueries: [{ query: ProfileDocument, variables: { handle } }],
  });

  useMemo(() => {
    (async () => {
      try {
        const response = await fetch(data.data.uri);
        const json = await response.json();
        setURIData(json);
      } catch (error) {
        displayError(error, enqueueSnackbar);
      }
    })();
  }, []);

  const handleEditProfile = async (e: any) => {
    e.preventDefault();

    try {
      await editProfileMutation({
        variables: {
          handle,
          avatar: uRIData?.image,
          data: {
            avatarMint: data.mint,
            avatarUpdateAuthority: data.updateAuthority,
          },
        },
      });

      enqueueSnackbar("Your profile has been updated ðŸ¥³.", {
        variant: "success",
      });
    } catch (err) {
      return displayError(err, enqueueSnackbar);
    }
  };

  return (
    <>
      <Box
        pb={4}
        pr={4}
        sx={{
          display: "flex",
          flexDirection: "column",
          minWidth: "140px",
          cursor: "pointer",
        }}
        key={uRIData?.image}
      >
        <Link
          onClick={() => toggleNftSelect(true)}
          color={theme.secondaryColor}
          underline="hover"
        >
          <Box>
            {uRIData?.image ? (
              <img src={uRIData?.image} alt={uRIData?.name} width="120" />
            ) : (
              <DoNotDisturbOnIcon fontSize="large" />
            )}
          </Box>
          <Box>
            <Typography sx={{ fontSize: "0.8em", maxWidth: "120px" }}>
              {uRIData?.name}
            </Typography>
          </Box>
        </Link>
      </Box>
      <Modal
        open={nftSelect}
        onClose={() => toggleNftSelect(false)}
        aria-labelledby="NFT action select"
        aria-describedby="Apply action for given NFT"
      >
        <Box
          sx={{
            position: "absolute",
            top: "40%",
            left: "50%",
            transform: "translate(-50%, -50%)",
            padding: 2,
            bgcolor: theme.background,
            border: `2px solid ${theme.blue.lightest}`,
            borderRadius: 1,
            maxWidth: "25em",
            width: "95%",
          }}
        >
          <Box
            display="flex"
            justifyContent="space-between"
            alignItems="flex-start"
          >
            <Typography id="nft-modal-title" variant="h6" component="h2">
              {uRIData?.name}
            </Typography>
            <IconButton
              aria-label="close-modal"
              onClick={() => toggleNftSelect(false)}
              sx={{
                padding: 0,
              }}
            >
              <Clear fontSize="small" sx={{ color: theme.blue.lighter }} />
            </IconButton>
          </Box>
          <Box display="flex" mt={2}>
            <Stack spacing={2}>
              {uRIData?.image && (
                <img src={uRIData?.image} alt={uRIData?.name} width="120" />
              )}
              {/* Verified */}
            </Stack>
            <Box
              ml={1}
              display="flex"
              sx={{
                overflow: "hidden",
                flexWrap: "wrap",
                alignContent: "flex-start",
              }}
            >
              {uRIData?.attributes && typeof uRIData?.attributes?.map === 'function' &&
                uRIData.attributes.map((nftItem) => (
                  <Chip
                    label={`${nftItem.trait_type}: ${nftItem.value}`}
                    variant="outlined"
                    size="small"
                    sx={{
                      margin: "0 0.5em 0.5em 0",
                      fontSize: "0.7em",
                    }}
                  />
                ))}
              <Chip
                component="a"
                target="_blank"
                href={`https://solana.fm/address/${data?.mint}?cluster=${cluster}-solana`}
                label="View in explorer"
                variant="outlined"
                size="small"
                color="primary"
                sx={{
                  margin: "0 0.5em 0.5em 0",
                  cursor: "pointer",
                  fontSize: "0.7em",
                }}
              />
            </Box>
          </Box>
          <Stack mt={2} spacing={1}>
            <Button
              fullWidth={true}
              variant="contained"
              onClick={handleEditProfile}
            >
              Set Profile Picture
            </Button>
            <Button
              fullWidth={true}
              variant="outlined"
              onClick={() => toggleNftSelect(false)}
            >
              Cancel
            </Button>
          </Stack>
        </Box>
      </Modal>
    </>
  );
};

export const NFTGallery: React.FC<ProfileQuery> = ({ profile }) => {
  const { connection } = useConnection();
  const { name } = currentCluster();
  const { enqueueSnackbar } = useSnackbar();

  const [nfts, setNfts] = useState<MetadataData[]>();
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    (async () => {
      try {
        setLoading(true);
        const nftMeta = profile.publicAddress
          ? await Metadata.findDataByOwner(connection, profile.publicAddress)
          : [];
        setNfts(nftMeta);
      } catch (error) {
        displayError(error, enqueueSnackbar);
      } finally {
        setLoading(false);
      }
    })();
  }, []);

  if (loading) return <Loader />;

  return (
    <>
      {nfts?.length ? (
        <Stack
          direction="row"
          sx={{
            flexWrap: "wrap",
            justifyContent: "flex-start",
            alignItems: "baseline",
          }}
        >
          {nfts &&
            nfts.map((nft: MetadataData) => (
              <NFTTile data={nft} key={nft.data.uri} cluster={name} />
            ))}
        </Stack>
      ) : (
        <Box pt={10} sx={{ display: "flex", justifyContent: "center" }}>
          <Typography>No NFTs in your wallet</Typography>
        </Box>
      )}
    </>
  );
};
