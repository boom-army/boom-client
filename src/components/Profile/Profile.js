import React from "react";
import { useQuery } from '@apollo/client';
import { useParams } from "react-router-dom";
import styled from "styled-components";
import ProfileInfo from "./ProfileInfo";
import { ShowTweet } from "../Tweet";
import { Loader } from "../Loader";
import { PROFILE } from "../../queries/profile";

const Wrapper = styled.div`
	padding-bottom: 5rem;

  .profile-top {
    display: flex;
    flex-direction: column;
    margin-left: 1rem;

    span.tweetsCount {
      margin-top: 0.1rem;
      color: ${(props) => props.theme.secondaryColor};
      font-size: 0.9rem;
    }
  }
`;

export const Profile = () => {
  const { handle } = useParams();

  const { loading, error, data } = useQuery(PROFILE, {
    variables: { handle },
  });

  if (loading) return <Loader />;

  return (
    <Wrapper>
      {/* <div className="profile-top">
        <span>{data && data.profile && data.profile.consumerName}</span>
        <span className="tweetsCount">
          {data && data.profile && data.profile.tweetsCount
            ? `${data.profile.tweetsCount} Tweets`
            : "No Tweets"}
        </span>
      </div> */}
      <ProfileInfo profile={data && data.profile} />
      {data && data.profile && data.profile.tweets && data.profile.tweets.length
        ? data.profile.tweets.map((tweet) => (
          <ShowTweet key={tweet.id} tweet={tweet} />
        ))
        : null}
    </Wrapper>
  );
};
