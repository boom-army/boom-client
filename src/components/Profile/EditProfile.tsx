import { Loader } from "../Loader";
import { EditProfileForm } from "./EditProfileForm";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")((props) => ({
  paddingBottom: "5rem",

  ".flex-wrapper": {
    display: "flex",
    justifyContent: "center",

    form: {
      ".cover-photo": {
        marginBottom: "1rem",
        cursor: "pointer",
      },

      ".avatar-input": {
        marginTop: "-100px",
        marginLeft: "1rem",
        cursor: "pointer",
      },

      "div.bio-wrapper": {
        background: props.theme.tertiaryColor2,
        marginBottom: "1.4rem",
        borderBottom: `1px solid ${props.theme.accentColor}`,
        padding: "0.5rem",

        label: {
          color: props.theme.secondaryColor,
          marginBottom: "0.4rem",
        },

        textarea: {
          fontSize: "1rem",
          width: "100%",
          background: props.theme.tertiaryColor2,
          border: "none",
          fontFamily: props.theme.font,
          color: props.theme.primaryColor,
        },
      },
    },
  },
  "@media screen and (max-width: 760px)": {
    ".flex-wrapper": {
      display: "block",
    },
  },
}));

export const EditProfile = ({ data, loading, setUser }: any) => {
  if (loading) return <Loader />;

  return (
    <Wrapper>
      <div className="flex-wrapper">
        <EditProfileForm profile={data && data?.profile} setUser={setUser} />
      </div>
    </Wrapper>
  );
};
