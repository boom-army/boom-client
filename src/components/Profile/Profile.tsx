import React, { useEffect, useState } from "react";
import ProfileInfo from "./ProfileInfo";
import { Box, Tab } from "@mui/material";
import { Loader } from "../Loader";
import { Meeps } from "./Meeps";
import { TabContext, TabPanel, TabList } from "@mui/lab";
import { useParams } from "react-router-dom";
import { useProfileLazyQuery } from "../../generated/graphql";
import { NFTGallery } from "./NFTGallery";
import { CustomResponse } from "../CustomResponse";

export const Profile: React.FC = () => {
  const [tabValue, setTabValue] = useState("1");

  let { handle } = useParams<string>();

  const [setHandle, { loading, data }] = useProfileLazyQuery();

  useEffect(() => {
    if (handle) setHandle({ variables: { handle } });
  }, [handle, setHandle]);

  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
    setTabValue(newValue);
  };

  if (loading) {
    return <Loader />;
  }
  if (!loading && !data) {
    return (
      <CustomResponse text="Oops, you are trying to visit a profile which doesn't exist. Make sure the profile handle exists" />
    );
  }

  return (
    <>
      <ProfileInfo profile={data && data.profile} />
      <Box sx={{ typography: "body1" }}>
        <TabContext value={tabValue}>
          <Box sx={{ borderBottom: 2, borderColor: "#374148" }}>
            <TabList
              onChange={handleChange}
              aria-label="Profile tablist select"
            >
              <Tab sx={{ minWidth: 150 }} label="NFT Gallery" value="1" />
              <Tab label="Meeps" value="2" />
            </TabList>
          </Box>
          <TabPanel value="1">
            {data && <NFTGallery publicAddress={data?.profile.publicAddress} />}
          </TabPanel>
          <TabPanel value="2">
            <Meeps data={data} />
          </TabPanel>
        </TabContext>
      </Box>
    </>
  );
};
