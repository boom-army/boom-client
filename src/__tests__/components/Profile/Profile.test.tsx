import { Profile } from "../../../components/Profile/Profile";
import renderer from "react-test-renderer";


describe("user profile : ", () => {
  test("doesn't exist", () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

