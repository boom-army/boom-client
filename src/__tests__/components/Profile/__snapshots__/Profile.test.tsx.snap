// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user profile :  doesn't exist 1`] = `
<div
  className="css-15inajd"
>
  <p>
    Oops, you are trying to visit a profile which doesn't exist. Make sure the profile handle exists
  </p>
</div>
`;


